src/storage/src/hummock/compactor/mod.rs

```diff
@@ -454,120 +477,179 @@ pub fn start_compactor(
+                                    )
+                                    .await
+                                    {
+                                        Ok((object_ids, total_object_count, total_object_size)) => {
+                                            Vacuum::report_full_scan_task(
+                                                object_ids,
+                                                total_object_count,
+                                                total_object_size,
+                                                meta_client,
+                                            )
+                                            .await;
+                                        }
+                                        Err(e) => {
+                                            tracing::warn!(error = %e.as_report(), "Failed to iter object");
                                         }
                                     }
-
-                                    ResponseEvent::PullTaskAck(_pull_task_ack) => {
-                                        // set flag
-                                        pull_task_ack.store(true, Ordering::SeqCst);
+                                });
+                            }
+                            ResponseEvent::ValidationTask(validation_task) => {
+                                executor.spawn(async move {
+                                    validate_ssts(validation_task, context.sstable_store.clone())
+                                        .await;
+                                });
+                            }
+                            ResponseEvent::CancelCompactTask(cancel_compact_task) => {
+                                if let Some(tx) = shutdown
+                                    .lock()
+                                    .unwrap()
+                                    .remove(&cancel_compact_task.task_id)
+                                {
+                                    if tx.send(()).is_err() {
+                                        tracing::warn!(
+                                            "Cancellation of compaction task failed. task_id: {}",
+                                            cancel_compact_task.task_id
+                                        );
                                     }
+                                } else {
+                                    tracing::warn!(
+                                        "Attempting to cancel non-existent compaction task. task_id: {}",
+                                            cancel_compact_task.task_id
+                                    );
                                 }
-                            });
+                            }
+
+                            ResponseEvent::PullTaskAck(_pull_task_ack) => {
+                                // set flag
+                                pull_task_ack.store(true, Ordering::SeqCst);
```

Comment:
Why is `pull_task_ack` a atomic? I think it can be a local var because it is only accessed in the event loop in a single-threaded manner.